kibana-reports/server/routes/utils/visual_report/visualReportHelper.ts

```diff
@@ -204,26 +204,44 @@ export const createVisualReport = async (
   return { timeCreated, dataUrl: buffer.toString('base64'), fileName };
 };
 
-const composeReportHtml = (
-  header: string,
-  footer: string,
-  screenshot: string
-) => {
-  const $ = cheerio.load(fs.readFileSync(`${__dirname}/report_template.html`), {
-    decodeEntities: false,
-  });
+const addReportStyle = async (page) => {
+  const css = fs.readFileSync(`${__dirname}/style.css`).toString();
 
-  $('.reportWrapper img').attr('src', `data:image/png;base64,${screenshot}`);
-  $('#reportingHeader > div.mde-preview > div.mde-preview-content').html(
-    header
-  );
-  if (footer === '') {
-    $('#reportingFooter').attr('hidden', 'true');
-  } else {
-    $('#reportingFooter > div.mde-preview > div.mde-preview-content').html(
-      footer
-    );
-  }
+  await page.evaluate((style) => {
+    const styleElement = document.createElement('style');
+    styleElement.innerHTML = style;
+    document.getElementsByTagName('head')[0].appendChild(styleElement);
+  }, css)
+}
 
-  return $.root().html() || '';
-};
+const addReportHeader = async (page, header) => {
+  const headerHtml = fs.readFileSync(`${__dirname}/header_template.html`)
+    .toString()
+    .replace('<!--CONTENT-->', header);
+
+  await page.evaluate((headerHtml: string) => {
+    const dashboardViewport = document.querySelector('#dashboardViewport');
```

Comment:
Need to handle visulization as well. Also we have defined variables for the id selectors for dashboard/visualization, please use them instead.
```
SELECTOR.dashboard
SELECTOR.visualization
```